### PR TITLE
un-i18n Modal Analytics

### PR DESCRIPTION
**Pull request by t3chguy**
with paired PR fixes https://github.com/vector-im/riot-web/issues/4549

also adds a couple of tracking statistics, boolean of whether a user is on matrix.org and vector.im HS and IS respectively.

paired with https://github.com/vector-im/riot-web/pull/4688

### CODE REVIEW

**Comment by t3chguy** [comment]
# Await Manual Testing

**Comment by lukebarnard1** [comment]
@t3chguy is the Manual Testing complete?

**Comment by lukebarnard1** [comment]
On looking at this, I'm not sure we should be leaking `e.message`... it could contain important stuff.

File: src/components/views/dialogs/SetEmailDialog.js [hunk 1]
```diff
@@ -106,7 +106,7 @@ export default React.createClass({
                 const QuestionDialog = sdk.getComponent("dialogs.QuestionDialog");
                 const message = _t("Unable to verify email address.") + " " +
                     _t("Please check your email and click on the link it contains. Once this is done, click continue.");
-                Modal.createDialog(QuestionDialog, {
+                Modal.createTrackedDialog('Verification Pending', 'M_THREEPID_AUTH_FAILED', QuestionDialog, {
```
Comment (lukebarnard1) [inline src/components/views/dialogs/SetEmailDialog.js]: `M_THREEPID_AUTH_FAILED` looks like it came from somewhere special. I'd go for something matching the others like plain english

**Comment by t3chguy** [comment]
@lukebarnard1 should we be gathering errcode or nothing at all then?

**Comment by lukebarnard1** [comment]
@t3chguy so following IRL with @dbkr, we reckon it'd be best to be paranoid for now and replace any sending of `e.message` or `e.toString()` with the empty string. We only need enough information to tell which dialog is popping up. If one dialog pops up enough to warrant investigation then we'd consider sending something else, but for now we're going to assume that all of our errors contain sensitive data (which is probably not true).

**Comment by t3chguy** [comment]
@lukebarnard1 alright, I'll get this changed today

File: src/ContentMessages.js [hunk 2]
```diff
@@ -360,7 +360,7 @@ class ContentMessages {
                     desc = _t('The file \'%(fileName)s\' exceeds this home server\'s size limit for uploads', {fileName: upload.fileName});
                 }
                 var ErrorDialog = sdk.getComponent("dialogs.ErrorDialog");
-                Modal.createDialog(ErrorDialog, {
+                Modal.createTrackedDialog('Upload failed', err.message, ErrorDialog, {
```
Comment (lukebarnard1) [inline src/ContentMessages.js]: err.message

File: src/Lifecycle.js [hunk 3]
```diff
@@ -240,7 +240,7 @@ function _handleRestoreFailure(e) {
     const SessionRestoreErrorDialog =
           sdk.getComponent('views.dialogs.SessionRestoreErrorDialog');
 
-    Modal.createDialog(SessionRestoreErrorDialog, {
+    Modal.createTrackedDialog('Session Restore Error', e.message, SessionRestoreErrorDialog, {
```
Comment (lukebarnard1) [inline src/Lifecycle.js]: e.message

File: src/components/structures/GroupView.js [hunk 4]
```diff
@@ -266,7 +266,7 @@ export default React.createClass({
             this.setState({uploadingAvatar: false});
             const ErrorDialog = sdk.getComponent("dialogs.ErrorDialog");
             console.error("Failed to upload avatar image", e);
-            Modal.createDialog(ErrorDialog, {
+            Modal.createTrackedDialog('Failed to upload image', e.toString(), ErrorDialog, {
```
Comment (lukebarnard1) [inline src/components/structures/GroupView.js]: e.toString

File: src/components/structures/GroupView.js [hunk 5]
```diff
@@ -288,7 +288,7 @@ export default React.createClass({
             });
             const ErrorDialog = sdk.getComponent("dialogs.ErrorDialog");
             console.error("Failed to save group profile", e);
-            Modal.createDialog(ErrorDialog, {
+            Modal.createTrackedDialog('Failed to update group', e.toString(), ErrorDialog, {
```
Comment (lukebarnard1) [inline src/components/structures/GroupView.js]: e.toString

File: src/components/structures/RoomView.js [hunk 6]
```diff
@@ -934,7 +934,7 @@ module.exports = React.createClass({
             }
             const ErrorDialog = sdk.getComponent("dialogs.ErrorDialog");
             console.error("Failed to upload file " + file + " " + error);
-            Modal.createDialog(ErrorDialog, {
+            Modal.createTrackedDialog('Failed to upload file', error.toString(), ErrorDialog, {
```
Comment (lukebarnard1) [inline src/components/structures/RoomView.js]: error.toString()

File: src/components/structures/RoomView.js [hunk 7]
```diff
@@ -1021,7 +1021,7 @@ module.exports = React.createClass({
         }, function(error) {
             var ErrorDialog = sdk.getComponent("dialogs.ErrorDialog");
             console.error("Search failed: " + error);
-            Modal.createDialog(ErrorDialog, {
+            Modal.createTrackedDialog('Search failed', error.toString(), ErrorDialog, {
```
Comment (lukebarnard1) [inline src/components/structures/RoomView.js]: error.toString()

File: src/components/structures/RoomView.js [hunk 8]
```diff
@@ -1217,7 +1217,7 @@ module.exports = React.createClass({
 
             var msg = error.message ? error.message : JSON.stringify(error);
             var ErrorDialog = sdk.getComponent("dialogs.ErrorDialog");
-            Modal.createDialog(ErrorDialog, {
+            Modal.createTrackedDialog('Failed to reject invite', error.toString(), ErrorDialog, {
```
Comment (lukebarnard1) [inline src/components/structures/RoomView.js]: error.toString()

File: src/components/structures/TimelinePanel.js [hunk 9]
```diff
@@ -923,7 +923,7 @@ var TimelinePanel = React.createClass({
             var message = (error.errcode == 'M_FORBIDDEN')
             	? _t("Tried to load a specific point in this room's timeline, but you do not have permission to view the message in question.")
                 : _t("Tried to load a specific point in this room's timeline, but was unable to find it.");
-            Modal.createDialog(ErrorDialog, {
+            Modal.createTrackedDialog('Failed to load timeline position', error.toString(), ErrorDialog, {
```
Comment (lukebarnard1) [inline src/components/structures/TimelinePanel.js]: error.toString()

File: src/components/structures/UserSettings.js [hunk 10]
```diff
@@ -335,7 +335,7 @@ module.exports = React.createClass({
         }, function(error) {
             const ErrorDialog = sdk.getComponent("dialogs.ErrorDialog");
             console.error("Failed to load user settings: " + error);
-            Modal.createDialog(ErrorDialog, {
+            Modal.createTrackedDialog('Can\'t load user settings', error.toString(), ErrorDialog, {
```
Comment (lukebarnard1) [inline src/components/structures/UserSettings.js]: error.toString()

File: src/components/structures/login/ForgotPassword.js [hunk 11]
```diff
@@ -152,7 +150,8 @@ module.exports = React.createClass({
 
     showErrorDialog: function(body, title) {
         var ErrorDialog = sdk.getComponent("dialogs.ErrorDialog");
-        Modal.createDialog(ErrorDialog, {
+        // TODO this will still lead to i18n in Analytics.
+        Modal.createTrackedDialog('Forgot Password Error', body, ErrorDialog, {
```
Comment (lukebarnard1) [inline src/components/structures/login/ForgotPassword.js]: Maybe just send "" here as well?

File: src/components/views/rooms/MemberInfo.js [hunk 12]
```diff
@@ -248,7 +248,7 @@ module.exports = withMatrixClient(React.createClass({
                     }, function(err) {
                         const ErrorDialog = sdk.getComponent("dialogs.ErrorDialog");
                         console.error("Kick error: " + err);
-                        Modal.createDialog(ErrorDialog, {
+                        Modal.createTrackedDialog('Failed to kick', err.message, ErrorDialog, {
```
Comment (lukebarnard1) [inline src/components/views/rooms/MemberInfo.js]: `err.message`

File: src/components/views/rooms/MemberInfo.js [hunk 13]
```diff
@@ -290,7 +290,7 @@ module.exports = withMatrixClient(React.createClass({
                     }, function(err) {
                         const ErrorDialog = sdk.getComponent("dialogs.ErrorDialog");
                         console.error("Ban error: " + err);
-                        Modal.createDialog(ErrorDialog, {
+                        Modal.createTrackedDialog('Failed to ban user', err.message, ErrorDialog, {
```
Comment (lukebarnard1) [inline src/components/views/rooms/MemberInfo.js]: err.message

File: src/components/views/rooms/MemberInfo.js [hunk 14]
```diff
@@ -340,7 +340,7 @@ module.exports = withMatrixClient(React.createClass({
                     console.log("Mute toggle success");
                 }, function(err) {
                     console.error("Mute error: " + err);
-                    Modal.createDialog(ErrorDialog, {
+                    Modal.createTrackedDialog('Failed to mute user', err.message, ErrorDialog, {
```
Comment (lukebarnard1) [inline src/components/views/rooms/MemberInfo.js]: err.message

File: src/components/views/rooms/MemberInfo.js [hunk 15]
```diff
@@ -385,7 +385,7 @@ module.exports = withMatrixClient(React.createClass({
                     dis.dispatch({action: 'view_set_mxid'});
                 } else {
                     console.error("Toggle moderator error:" + err);
-                    Modal.createDialog(ErrorDialog, {
+                    Modal.createTrackedDialog('Failed to toggle moderator status', err.message, ErrorDialog, {
```
Comment (lukebarnard1) [inline src/components/views/rooms/MemberInfo.js]: err.message

File: src/components/views/rooms/MemberInfo.js [hunk 16]
```diff
@@ -406,7 +406,7 @@ module.exports = withMatrixClient(React.createClass({
             }, function(err) {
                 const ErrorDialog = sdk.getComponent("dialogs.ErrorDialog");
                 console.error("Failed to change power level " + err);
-                Modal.createDialog(ErrorDialog, {
+                Modal.createTrackedDialog('Failed to change power level', err.message, ErrorDialog, {
```
Comment (lukebarnard1) [inline src/components/views/rooms/MemberInfo.js]: err.message

File: src/components/views/rooms/MessageComposerInput.js [hunk 17]
```diff
@@ -721,15 +721,15 @@ export default class MessageComposerInput extends React.Component {
...
+                    Modal.createTrackedDialog('Server error', '', ErrorDialog, {
                         title: _t("Server error"),
                         description: ((err && err.message) ? err.message : _t("Server unavailable, overloaded, or something else went wrong.")),
                     });
                 });
             } else if (cmd.error) {
                 console.error(cmd.error);
                 const ErrorDialog = sdk.getComponent("dialogs.ErrorDialog");
-                Modal.createDialog(ErrorDialog, {
+                Modal.createTrackedDialog('Command error', cmd.error, ErrorDialog, {
```
Comment (lukebarnard1) [inline src/components/views/rooms/MessageComposerInput.js]: cmd.error

File: src/createRoom.js [hunk 18]
```diff
@@ -115,7 +115,7 @@ function createRoom(opts) {
             action: 'join_room_error',
         });
         console.error("Failed to create room " + roomId + " " + err);
-        Modal.createDialog(ErrorDialog, {
+        Modal.createTrackedDialog('Failure to create room', err.message, ErrorDialog, {
```
Comment (lukebarnard1) [inline src/createRoom.js]: err.message

**Pull request merged**